src/web/app/pages-instructor/instructor-track-view-page/instructor-track-view-page.component.ts

```diff
@@ -137,34 +141,35 @@ export class InstructorTrackViewPageComponent implements OnInit {
         intent: Intent.INSTRUCTOR_RESULT,
       })
       .subscribe((feedbackSession: FeedbackSession) => {
-        if (feedbackSession.resultVisibleFromTimestamp) {
-          this.publishedTime = feedbackSession.resultVisibleFromTimestamp;
-        }
+        this.publishedTime = feedbackSession.resultVisibleFromTimestamp || this.publishedTime;
 
-        if (!(feedbackSession.publishStatus === FeedbackSessionPublishStatus.PUBLISHED)) {
-          // Feedback session is not published, do not need to search.
+        // Feedback session is not published, do not need to search.
+        if (feedbackSession.publishStatus !== FeedbackSessionPublishStatus.PUBLISHED) {
           this.isSearching = false;
           this.statusMessageService.showErrorToast('This feedback session is not published');
-          // this.emptyResult();
-        } else if (this.publishedTime < Date.now() - this.LOGS_RETENTION_PERIOD * 24 * 60 * 60 * 1000) {
-          // Feedback session is published more than 30 days ago, open a dialog.
-          this.openModal(feedbackSession);
-        } else {
-          // Published time of the feedback session is less than 30 days ago, update the form model to
-          // search only up till the publish date.
-          const publishedDate: Date = new Date(this.publishedTime);
-          this.notViewedSince = publishedDate.getTime();
-          this.logsDateFrom = {
-            year: publishedDate.getFullYear(),
-            month: publishedDate.getMonth() + 1,
-            day: publishedDate.getDate(),
-          };
-          this.logsTimeFrom = {
-            hour: publishedDate.getHours(),
-            minute: publishedDate.getMinutes(),
-          };
-          this.search();
+          return;
         }
+
+        // Feedback session is published more than 30 days ago, open a dialog.
+        if (this.publishedTime < Date.now() - this.LOGS_RETENTION_PERIOD_IN_MILLISECONDS) {
```

Comment:
When I read this, personally it doesn't feel particularly fluent:( in particular `Date.now()` minus `some_milliseconds`.
Is there a better way to express this?

Reply:
Just discussing

Reply:
Would it be better if I change to
```
const earliestSearchDateTimestamp = Date.now() - this.LOGS_RETENTION_PERIOD_IN_MILLISECONDS;
if(this.publishedTime < earliestSearchDateTimestamp) {...}
```